Sources/BraintreeCore/BTHTTP.swift

```diff
@@ -393,19 +405,18 @@ class BTHTTP: NSObject, URLSessionTaskDelegate {
 
     func handleJSONResponseError(
         json: BTJSON,
-        response: URLResponse,
-        completion: @escaping (Error?) -> Void
-    ) {
+        response: URLResponse
+    ) async throws -> Error? {
```

Comment:
Same for this function. There would be no async work, so we shouldn't need the async keyword.

We could also refactor this function to `throw` an error instead of having it as a return type. This would follow the async/await pattern a bit closer, since we'd eventually have to throw the returned error.